### PR TITLE
Slash Commands & Buttons

### PR DESCRIPTION
**Pull request by milanmdev**
Slash commands & buttons are being added.

This will be a draft until I finish everything that I plan to do.

Also give me direct commit access :p

### CODE REVIEW

**Comment by milanmdev** [comment]
What I added:
- buttons to the following commands: `support, stats, invite, privacy`
- slash commands for the following commands: `support, invite, privacy, songoftheday`

In the config.json, you will need to add the following new fields:
```
"applicationID": "",
"publicKey": ""
```

**Review by taynotfound — COMMENTED** [review]
Seems pretty good

**Comment by ImFelixprogram** [comment]
@tayron1 Please do not merge unless you viewed through all code, there's some changes to be made.

File: README.md [hunk 1]
```diff
@@ -114,26 +114,8 @@ This Bot is a Modified fork from [Tomato6966/discord-js-lavalink-Music-Bot-erela
...
-@ECHO Starting BOT
-ECHO ==========================
-start cmd /k node .
-exit /s'
-```
-Windows `start.bat` file
-Which starts Lavalink and the Bot together via one click ;)
-
-Have it in the Same folder as `index.js` and `Lavalink.jar`
+# Start Command
```
Comment (ImFelixprogram) [inline README.md:117]: Lavalink has still to be started beforehand but you left that out with this edit. Also, what you state here is also mentioned in the section **Selfhosting**, where you should have changed that [Step 7]. This section which you removed was a all-in-one start file template.

File: commands/🔰 Info/support.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ module.exports = {
         .setTitle(":heart: Join here!")
         .setFooter(ee.footertext, ee.footericon)
         .setURL(`https://discord.gg/kUuNQwnvCF`)
-        .setDescription(`[Click here](https://discord.gg/kUuNQwnvCF)`)
+        .setDescription(`[Click here](https://discord.gg/kUuNQwnvCF) (https://discord.gg/kUuNQwnvCF)`)
```
Comment (ImFelixprogram) [inline commands/🔰 Info/support.js]: The link address is shown when you click on it in Discord, no need to add it afterwards

File: index.js [hunk 3]
```diff
@@ -1,32 +1,40 @@
...
-  restWsBridgetimeout: 100,
-  disableEveryone: true,
-  partials: ['MESSAGE', 'CHANNEL', 'REACTION'],
+    restTimeOffset: 0,
+    restWsBridgetimeout: 100,
+    disableEveryone: true,
+    partials: ['MESSAGE', 'CHANNEL', 'REACTION'],
 });
+const creator = new SlashCreator({
+    applicationID: require("./botconfig/config.json").applicationID,
```
Comment (ImFelixprogram) [inline index.js:33]: Require once before creating the const creator to avoid requiring the same file three times

File: slash-commands/support.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    async run(ctx) {
+        await ctx.defer();
+
+        let embed = new Discord.MessageEmbed()
+            .setColor(ee.color)
+            .setTitle(":heart: Join here!")
+            .setFooter(ee.footertext, ee.footericon)
+            .setURL(`https://discord.gg/kUuNQwnvCF`)
+            .setDescription(`[Click here](https://discord.gg/kUuNQwnvCF) (https://discord.gg/kUuNQwnvCF)`)
```
Comment (ImFelixprogram) [inline slash-commands/support.js:22]: Same here, Discord shows what link you are being redirected to as you click on it

File: commands/🔰 Info/support.js [hunk 5]
```diff
@@ -10,33 +11,41 @@ module.exports = {
...
-        .setURL(`https://discord.gg/kUuNQwnvCF`)
-        .setDescription(`[Click here](https://discord.gg/kUuNQwnvCF)`)
-        .setImage("https://cdn.discordapp.com/attachments/802144342185738250/820385232686546945/cover.png")
+        try {
+            let embed = new Discord.MessageEmbed()
+                .setColor(ee.color)
+                .setTitle(":heart: Join here!")
+                .setFooter(ee.footertext, ee.footericon)
+                .setURL(`https://discord.gg/kUuNQwnvCF`)
+                .setDescription(`[Click here](https://discord.gg/kUuNQwnvCF) (https://discord.gg/kUuNQwnvCF)`)
```
Comment (ImFelixprogram) [inline commands/🔰 Info/support.js:20]: Discord shows what URL you are opening when clicking on it, there's no need to append it afterwards.

**Comment by taynotfound** [comment]
I looked at everything at it seemed okay and it's working good @ImFelixprogram

**Review by ImFelixprogram — COMMENTED** [review]
_Please complete the documentation and maybe change some of the code based on the comments_ - Based on the fact that Tayron already merged, you don't have to do that, I will make a commit changing that, thanks for your pull request.

**Comment by ImFelixprogram** [comment]
@tayron1 We also need to pay some attention to code cleanliness and complete documentation, please wait for another review next time before merging

**Comment by taynotfound** [comment]
Aha mk

**Pull request merged**